### ISSUE TITLE
deposit facet before smart wallet is provisioned

### ISSUE THREAD

**Issue by dckc** [Agoric/Agoric#10640]
## What is the Problem Being Solved?

Contracts can [use namesByAddress to send payments](https://docs.agoric.com/guides/integration/name-services.html#namesbyaddress-namesbyaddressadmin-and-depositfacet-per-account-namespace), but only for accounts that have provisioned a smartwallet.

There are work-arounds using `namesByAddressAdmin` or `bankManager`, but those are both are excess authority.

The Tribbles project seems to be facing this issue as part of...
 - https://github.com/hindley-milner-systems/dapp-ertp-airdrop/issues/141

Related problems:

 - #8594
 - #8378

## Description of the Design

As discussed in https://github.com/hindley-milner-systems/dapp-ertp-airdrop/issues/141#issuecomment-2524193433

The platform should provide some sort of `getDepositFacet(addr, brand)`. For brands that are known to be in the vbank, using `bankManager` seems straightforward:

```js
const attenuateBankManager(bankManager) {
  const getDepositFacet = (addr, brand) => E(E(E(bankManager).getBankForAddress(addr)).getPurseForBrand(brand)).getDepositFacet();
  return getDepositFacet;
}
```

Otherwise, an analagous facility based on `namesByAddressAdmin` seems in order.

## Security Considerations

yes :)

## Scaling Considerations

unilaterally provisioning smart wallets seems expensive. (that's why the approach based on `bankManager` is preferable)

## Test Plan

TBD

## Upgrade Considerations

It's probably straightforward to add this to the bootstrap promise space, though investing further in the bootstrap vat when its upgrade path isn't clear (#8849 ) is tricky.

cc @michaelfig

**Comment by michaelfig** [comment]
> There are work-arounds using `namesByAddressAdmin` or `bankManager`, but those are both are excess authority.

It seems that the workarounds could be structured so that they don't expose `namesByAddressAdmin` or `bankManager` and are created as durable objects exported from a non-bootstrap vat and added to the promise space.

Something like:

```js
// Idempotent initialization.
const bankDepositor = await E(bankManager).getBankDepositor(); // obtain the singleton bankDepositor.
produce.bankDepositor.resolve(bankDepositor);

// Get a specific deposit facet.
const myDepositFacet = E(consume.bankDepositor).getDepositFacet(addr, brand);
```

Would that and/or its `namesByAddressAdmin` equivalent satisfy your requirement not to add more non-durable functions to the promise space and export the durable objects only from non-bootstrap vats?

**Comment by dckc** [comment]
Yeah... It would involve upgrading the bank / provisioning vat, but that should be feasible.
